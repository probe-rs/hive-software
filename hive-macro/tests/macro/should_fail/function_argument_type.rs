#[hive_macro::hive_test]
fn test_function(test_channel: &mut String, handle: String) {}

fn main() {}
