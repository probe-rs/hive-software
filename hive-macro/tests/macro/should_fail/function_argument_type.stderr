error: Wrong function argument type

         = help: Only the argument types listed in the example below are valid types to use in a Hive testfunction.
         = note: example: fn my_fancy_test(test_channel: &mut dyn TestChannelHandle, session: &mut Session, target_info: &HiveTargetInfo);

 --> tests/macro/should_fail/function_argument_type.rs:2:37
  |
2 | fn test_function(test_channel: &mut String, handle: String, info: String) {}
  |                                     ^^^^^^
